<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>KernelDialog</class>
 <widget class="QDialog" name="KernelDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>479</width>
    <height>303</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <widget class="QTableWidget" name="kernelTable"/>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="colsLabel">
         <property name="text">
          <string>Columns:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="rowsLabel">
         <property name="text">
          <string>Rows:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="rowsBox">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>50</number>
         </property>
         <property name="value">
          <number>3</number>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="colsBox">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>50</number>
         </property>
         <property name="singleStep">
          <number>1</number>
         </property>
         <property name="value">
          <number>3</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QGridLayout" name="gridLayout">
       <item row="1" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Factor:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QDoubleSpinBox" name="scaleFactorBox">
         <property name="maximum">
          <double>500.000000000000000</double>
         </property>
         <property name="value">
          <double>1.000000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QPushButton" name="normalizeButton">
       <property name="text">
        <string>Normalize</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Ignored" vsizetype="Maximum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>KernelDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>466</x>
     <y>239</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>KernelDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>466</x>
     <y>245</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>normalizeButton</sender>
   <signal>clicked()</signal>
   <receiver>KernelDialog</receiver>
   <slot>changeNormalizeFactor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>426</x>
     <y>145</y>
    </hint>
    <hint type="destinationlabel">
     <x>437</x>
     <y>164</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>colsBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>KernelDialog</receiver>
   <slot>changeMatrixSize()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>433</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>45</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rowsBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>KernelDialog</receiver>
   <slot>changeMatrixSize()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>430</x>
     <y>56</y>
    </hint>
    <hint type="destinationlabel">
     <x>471</x>
     <y>81</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>kernelTable</sender>
   <signal>cellChanged(int,int)</signal>
   <receiver>KernelDialog</receiver>
   <slot>updateMatrix()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>228</x>
     <y>161</y>
    </hint>
    <hint type="destinationlabel">
     <x>216</x>
     <y>294</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>changeNormalizeFactor()</slot>
  <slot>changeMatrixSize()</slot>
  <slot>updateMatrix()</slot>
 </slots>
</ui>
